app/validators/special_character_validator.rb

```diff
@@ -1,17 +1,13 @@
 class SpecialCharacterValidator < ActiveModel::EachValidator
 
   def validate_each(record, attribute, value)
-    return if value.nil?
+    return if value.nil? || value.blank?
 
-    if allow_comma == true && allow_numbers == true
-      regex = /^[a-zA-Z0-9 ,'-]+$/i
-    elsif allow_comma == true
-      regex = /^[a-zA-Z ,'-]+$/i
-    elsif allow_numbers == true
-      regex = /^[a-zA-Z0-9 '-]+$/i
-    else
-      regex = /^[a-zA-Z '-]+$/i
-    end
+    allowed_values = "a-zA-Z -'"
```

Comment:
strings are frozen in later versions of ruby - and when we have #frozen_string_literals=true which we should have in our ruby files

Therefore the correct way of doing this is to use an array of strings - then use the .join method at the end to join the strings together